### PR TITLE
Cleanup / Simplify `RuleConfig` trait

### PR DESCRIPTION
**Pull request by fuchsnj**
This cleans up the `RuleConfig` trait a bit. The builder / config structs were merged into one since they were identical. The scanner has standardized on accepting `Arc<dyn RuleConfig>` instead of accepting generic rule configs, which simplifies things a bit and will be used soon in a future PR updating the go bindings.

There were also some minor refactoring to split the code into more files.

This is a breaking change, and I'll open a PR downstream fixing any breakages before merging this in. No actual business logic was changed.

### CODE REVIEW

File: sds/src/scanner/regex_rule/mod.rs [hunk 1]
```diff
@@ -0,0 +1,29 @@
+pub mod compiled;
+pub mod config;
+//
```
Comment (vinckama) [inline sds/src/scanner/regex_rule/mod.rs:3]: Would it be ok to remove this comment?

**Review by vinckama — COMMENTED** [review]
I added a few comments/questions

File: sds/src/scanner/config.rs [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    pub included_keywords: Vec<String>,
+
+    #[serde_as(deserialize_as = "DefaultOnNull")]
+    #[serde(default)]
+    pub excluded_keywords: Vec<String>,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
+#[serde(tag = "type")]
+pub enum SecondaryValidator {
```
[the remark this answers is not in the document]
Comment (fuchsnj) [reply sds/src/scanner/config.rs:34]: Good point

File: sds/src/scanner/regex_rule/mod.rs (hunk 1 above)
Comment (fuchsnj) [reply sds/src/scanner/regex_rule/mod.rs]: done https://github.com/DataDog/dd-sensitive-data-scanner/pull/85/commits/ffc0fad9d7bd59165d2b75ad63d31a4f69429827

File: sds/src/scanner/mod.rs [hunk 3]
```diff
@@ -1,28 +1,28 @@
...
+use crate::{CreateScannerError, EncodeIndices, MatchAction, Path};
 use regex_automata::meta::Regex as MetaRegex;
 use std::sync::Arc;
 
 use self::cache_pool::{CachePool, CachePoolBuilder, CachePoolGuard};
-use self::metrics::RuleMetrics;
 use self::metrics::ScannerMetrics;
-use crate::proximity_keywords::compile_keywords_proximity_config;
-use crate::scanner::regex_rule::RegexCompiledRule;
+// use crate::scanner::regex_rule::RegexCompiledRule;
```
[the remark this answers is not in the document]
Comment (fuchsnj) [reply sds/src/scanner/mod.rs]: done

File: sds/src/scanner/config.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (fuchsnj) [reply sds/src/scanner/config.rs]: https://github.com/DataDog/dd-sensitive-data-scanner/pull/85/commits/ffc0fad9d7bd59165d2b75ad63d31a4f69429827

File: sds/src/config/mod.rs [hunk 4]
```diff
@@ -0,0 +1 @@
+
```
[the remark this answers is not in the document]
Comment (fuchsnj) [reply sds/src/config/mod.rs]: removed

**Review by vinckama — APPROVED** [review]

**Pull request merged**